src/Storages/Statistics/Statistics.h

```diff
@@ -46,6 +45,8 @@ class IStatistics
     virtual Float64 estimateEqual(const Field & val) const; /// cardinality of val in the column
     virtual Float64 estimateLess(const Field & val) const;  /// summarized cardinality of values < val in the column
 
+    virtual DataTypePtr getTargetType() const { return nullptr; }
```

Comment:
To the reader, it is not clear what the "target type", respectively `getTargetType`, represents. Please add a comment. The comment ideally also describes what `nullptr` means as a target type.